enterprise/api_client/client.py

```diff
@@ -0,0 +1,125 @@
+"""
+Base API clients to comunicate with edx services.
+"""
+import logging
+from functools import wraps
+from time import time
+from urllib.parse import urljoin
+
+import requests
+from edx_rest_api_client.auth import SuppliedJwtAuth
+from edx_rest_api_client.client import OAuthAPIClient
+
+from django.apps import apps
+from django.conf import settings
+
+from enterprise.utils import NotConnectedToOpenEdX
+
+try:
+    from openedx.core.djangoapps.oauth_dispatch import jwt as JwtBuilder
+except ImportError:
+    JwtBuilder = None
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class APIClientMixin:
+    """
+    API client mixin.
+    """
+    APPEND_SLASH = False
+    API_BASE_URL = ""
+
+    def get_api_url(self, path):
+        """
+        Helper method to construct the API URL.
+
+        Args:
+            path (str): the API endpoint path string.
+        """
+        return urljoin(f"{self.API_BASE_URL}/", f"{path.strip('/')}{'/' if self.APPEND_SLASH else ''}")
+
+
+class BackendServiseAPIClient(APIClientMixin):
```

Comment:
In this name, Service is misspelled as `Servise`. All subclasses should be updated with the correct spelling as well.

Reply:
Fixed, thanks